ttnn/cpp/ttnn/graph_processor.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include "tt_metal/graph_tracking.hpp"
+#include "third_party/json/json.hpp"
+#include <mutex>
+#include <stack>
+#include <typeindex>
+#include <unordered_map>
+#include <functional>
+#include <any>
+
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace ttnn {
+
+    class ProcessorHooks : public tt::tt_metal::IGraphHooks {
+    private:
+        bool do_block = false;
+    public:
+        ProcessorHooks() = default;
+        virtual bool hook_allocate(tt::tt_metal::Buffer* buffer, bool bottom_up) override;
```

Comment:
The general guideline IIRC is to use only `override` for overrides and drop the `virtual`, which is what you do below.

Reply:
agree